src/sidebar/components/test/annotation-share-control-test.js

```diff
@@ -0,0 +1,185 @@
+  });
+
+  it('toggles the share control element when the button is clicked', () => {
+    const wrapper = createComponent();
+
+    act(() => {
+      wrapper
+        .find('AnnotationActionButton')
+        .props()
+        .onClick();
+    });
+    wrapper.update();
+
+    assert.isTrue(wrapper.find('.annotation-share-panel').exists());
+  });
+
+  it('renders the share URI in a readonly input field', () => {
+    const wrapper = createComponent();
+    openElement(wrapper);
+
+    const inputEl = wrapper.find('input');
+    assert.equal(inputEl.prop('value'), fakeShareUri);
+    assert.isTrue(inputEl.prop('readOnly'));
+  });
+
+  describe('copying the share URI to the clipboard', () => {
+    it('copies the share link to the clipboard when the copy button is clicked', () => {
+      const wrapper = createComponent();
+      openElement(wrapper);
+
+      wrapper.find('.annotation-share-panel__copy-btn').simulate('click');
+
+      assert.calledWith(
+        fakeCopyToClipboard.copyText,
+        'https://www.example.com'
+      );
+    });
+
+    it('confirms link copy when successful', () => {
+      const wrapper = createComponent();
+      openElement(wrapper);
+
+      wrapper.find('.annotation-share-panel__copy-btn').simulate('click');
+
+      assert.calledWith(fakeFlash.info, 'Copied share link to clipboard');
+    });
+
+    it('flashes an error if link copying unsuccessful', () => {
+      fakeCopyToClipboard.copyText.throws();
+      const wrapper = createComponent();
+      openElement(wrapper);
+
+      wrapper.find('.annotation-share-panel__copy-btn').simulate('click');
+
+      assert.calledWith(fakeFlash.error, 'Unable to copy link');
+    });
+  });
+
+  [
+    {
+      groupType: 'private',
+      isPrivate: false,
+      expected: 'Only members of the group My Group may view this annotation.',
+    },
+    {
+      groupType: 'open',
+      isPrivate: false,
+      expected: 'Anyone using this link may view this annotation.',
+    },
+    {
+      groupType: 'private',
+      isPrivate: true,
+      expected: 'Only you may view this annotation.',
+    },
+    {
+      groupType: 'open',
+      isPrivate: true,
+      expected: 'Only you may view this annotation.',
+    },
+  ].forEach(testcase => {
+    it(`renders the correct sharing information for a ${testcase.groupType} group when annotation privacy is ${testcase.isPrivate}`, () => {
+      fakeGroup.type = testcase.groupType;
+      const wrapper = createComponent({ isPrivate: testcase.isPrivate });
+      openElement(wrapper);
+
+      const permissionsEl = wrapper.find(
+        '.annotation-share-panel__permissions'
+      );
+      assert.equal(permissionsEl.text(), testcase.expected);
+    });
+  });
+
+  it('focuses the share-URI input when opened', () => {
+    document.body.focus();
+
+    const wrapper = createComponent();
+    openElement(wrapper);
+    wrapper.update();
+
+    assert.equal(
+      document.activeElement.getAttribute('aria-label'),
```

Comment:
I could not get this to work, so you need to call document.body.focus() first? Maybe a stackoverlow q/a?

Reply:
I'm not following this comment :). Are you saying you couldn't get this kind of test to work in another component?

Reply:
Yes, in another test, I tried setting the focus on an input, then checking to see the value of document.activeElement, but it never worked. Here you set the focus on the body first via `document.body.focus`, then clicked a button and `document.activeElement` seems to work.